packages/metal-quartz-collapse/src/MetalQuartzCollapse.js

```diff
@@ -0,0 +1,273 @@
+	 * of the transitionend event; returns false if it's not supported
+	 * @protected
+	 */
+	getTransitionEndEvent_() {
+		let el = document.createElement('metalQuartzTransitionEnd');
+
+		let transitionEndEvents = {
+			transition: 'transitionend',
+			WebkitTransition: 'webkitTransitionEnd',
+			MozTransition: 'transitionend',
+			OTransition: 'oTransitionEnd otransitionend'
+		};
+
+		for (let name in transitionEndEvents) {
+			if (el.style[name] !== undefined) {
+				return transitionEndEvents[name];
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Handles a `click` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleClick_(event) {
+		this.toggle();
+	}
+
+	/**
+	 * Handles a `keydown` event on the headers.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleKeydown_(event) {
+		if (event.keyCode === 13 || event.keyCode === 32) {
+			this.toggle();
+			event.preventDefault();
+		}
+	}
+
+	/**
+	 * Handles the `transitionend` event on the content.
+	 * @param {!Event} event
+	 * @protected
+	 */
+	handleTransitionEnd_(event) {
+		this.content.transitionType ? this.open() : this.close();
+
+		this.transitionEventHandler_.removeAllListeners();
+	}
+
+	/**
+	 * Adds CSS classes and properties to the `content` element when `collapsed`
+	 * is false
+	 */
+	open() {
+		dom.addClasses(this.content, this.closedClass);
+		dom.addClasses(this.content, this.openClass);
+		dom.removeClasses(this.content, this.transitionClass);
+		this.content.setAttribute('aria-expanded', true);
+		this.content.style.removeProperty('height');
+	}
+
+	/**
+	 * Fires a synthetic `transitionend` event for browsers that don't support
+	 * CSS3 transitions
+	 * @param {!Element} element
+	 * @protected
+	 */
+	shimUnsupportedTransition_(element) {
+		if (!this.supportsTransitionEnd) {
+			anim.emulateTransitionEnd(element);
+		}
+	}
+
+	/**
+	 * Syncs the `content` element according to the value of the `collapsed`
+	 * state, attaching and removing css properties and classes needed to open
+	 * and close the element.
+	 */
+	syncCollapsed() {
+		this.collapsed ? this.animateClose() : this.animateOpen();
+	}
+
+	/**
+	 * Syncs the component according to the value of the `headers` state,
+	 * attaching events to the new element and detaching from any previous one.
+	 */
+	syncHeaders() {
+		this.eventHandler_.removeAllListeners();
+
+		if (Array.isArray(this.headers)) {
+			for (let i = 0; i < this.headers.length; i++) {
+				this.eventHandler_.add(
+					dom.on(this.headers[i], 'click', this.handleClick_.bind(this)),
+					dom.on(this.headers[i], 'keydown', this.handleKeydown_.bind(this))
+				);
+			}
+		} else {
+			this.eventHandler_.add(
+				dom.on(this.headers, 'click', this.handleClick_.bind(this)),
+				dom.on(this.headers, 'keydown', this.handleKeydown_.bind(this))
+			);
+		}
+	}
+
+	/**
+	 * Toggles the content's visibility.
+	 */
+	toggle() {
+		this.transitionEventHandler_.add(
```

Comment:
We don't want open_() and close_() to run on every transitionend. We will run into problems if there is another kind of transition added to the content. It will open or close the menu after that transition as well.

From a performance point of view, I don't know if having X transitionend listeners on X numbers of content is actually better than adding and removing one on toggle.